velox/dwio/common/compression/Compression.cpp

```diff
@@ -293,6 +293,110 @@ std::pair<int64_t, bool> ZstdDecompressor::getDecompressedLength(
   return {uncompressedLength, true};
 }
 
+class GzipIAADecompressor : public AsyncDecompressor {
```

Comment:
It will be good to split the file for different codec, like CompressionZlib, CompressionZSTD, compressionZlibIAA, etc. and put the QPLJobPool.cpp/h in side of compressionZLibIAA. The logic can be much clean.

We also have the plan to add QAT codec in parquet read/write. It will be another code file.

@pedroerp your thought?

Reply:
It is hard to move this to QPL library. Qpl job pool is customized struct and the retry mechanism after qpl_submit_job  returns failure is better defined by workloads.

Reply:
these details should be hidden into the codec itself